teams.md

```diff
@@ -215,3 +190,18 @@ on:
   improving our security posture
   audit trail for generated and published assets and ensuring we're using
   encryption at rest in the right places.
+
+## Security
```

Comment:
Is it worth talking about how this is different from Product Security team (I dislike the team name)? Or perhaps within Product Security, we talk about how there's a separate Security team?

Reply:
Yes -- I'll cross link between the two sections.

Reply:
Cross linked and took a shot at explaining the differences in https://github.com/gravitational/careers/pull/66/commits/1f4cc986761af9e6db5017e5de31f1e468d76e0a